pandas/core/internals/concat.py

```diff
@@ -198,6 +198,27 @@ def concatenate_managers(
     if isinstance(mgrs_indexers[0][0], ArrayManager):
         return _concatenate_array_managers(mgrs_indexers, axes, concat_axis, copy)
 
+    new_mgrs_indexers = []
```

Comment:
i'd feel better if you make this a separate function and just call it here

Reply:
so just refactor out L201-L220 as a new function?  sure